numba/core/entrypoints.py

```diff
@@ -1,6 +1,9 @@
 import logging
 
-from pkg_resources import iter_entry_points
+try:
+    from importlib import metadata as importlib_metadata
+except ImportError:
+    import importlib_metadata
```

Comment:
What if this fails? I'm wondering if there ought to be an explicit Python>=3.8 branch which just imports it and then another which tries and fails gracefully if the package is missing.

Reply:
It will only fail on Python < 3.8 if importlib_metadata is missing (it is listed as requirement).

Reply:
We tend to assume nothing about how Numba was built, it's entirely possible someone just grabbed the source and some limited set of bootstrap packages available for their unusual architecture and built it (this actually happens in practice, we see these people on the issue tracker when strange things happen!). As a result, we tend to err on the side of trying to be as helpful as possible and catch things like this and say something about it being needed.

Further, this will need adding to the dependencies section in the docs: https://github.com/numba/numba/blob/13ece9b97e6f01f750e870347f231282325f60c3/docs/source/user/installing.rst#dependency-list